### PR TITLE
Allow excluding Maven dependency from tree

### PR DESCRIPTION
**Pull request by yahavi**
- [x] I signed [JFrog's CLA](https://secure.echosign.com/public/hostedForm?formid=5IYKLZ2RXB543N).
- [x] All [tests](https://github.com/jfrog/jfrog-vscode-extension#building-and-testing-the-sources) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `npm run format` for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: src/main/exclusions/abstractExclusion.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ */
+export abstract class AbstractExclusion {
+    constructor(private _pkgType: string) {}
+
+    public isMatched(dependenciesTreeNode: DependenciesTreeNode): boolean {
+        return dependenciesTreeNode.generalInfo.pkgType === this._pkgType;
+    }
+
+    /**
+     * Show the dependency in the project descriptor (i.e package.json) file after right click on the components tree and a left click on "Show in project descriptor".
```
Comment (Or-Gabay) [inline src/main/exclusions/abstractExclusion.ts:14]: Incorrect comment.

File: src/main/utils/mavenUtils.ts [hunk 2]
```diff
@@ -73,6 +74,24 @@ export class MavenUtils {
         return [];
     }
 
+    public static getDependency(document: vscode.TextDocument, dependenciesTreeNode: DependenciesTreeNode | undefined): string {
```
Comment (Or-Gabay) [inline src/main/utils/mavenUtils.ts:77]: It looks like this code is duplicate, export it to a different function.

File: src/main/exclusions/mavenExclusion.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        });
+        await textDocument.save();
+    }
+
+    /**
+     * Return true if the exclusion is already exist in pom. This cover the case of many exclusions under <exclusions>...</exclusions> tag.
+     * @param exclusions - The candidate exclusion
+     * @param groupId - The group ID
+     * @param artifactId - The artifact ID
+     */
```
Comment (eyalbe4) [inline src/main/exclusions/mavenExclusion.ts:142]: isExclusionExist --> isExclusionExists

File: src/main/exclusions/mavenExclusion.ts [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+            .replace(/^(\s*)/, '');
+
+        let textEdit: vscode.TextEditor = await vscode.window.showTextDocument(textDocument);
+        await textEdit.edit(textEdit => {
+            textEdit.replace(new vscode.Range(startPos, endPos), dependencyWithExclusionStr);
+        });
+        await textDocument.save();
+    }
+
+    /**
```
Comment (eyalbe4) [inline src/main/exclusions/mavenExclusion.ts:137]: ...already exist... --> ...already exists...

...cover... --> ...covers...

File: src/main/utils/mavenUtils.ts [hunk 5]
```diff
@@ -73,6 +69,33 @@ export class MavenUtils {
         return [];
     }
 
+    /**
```
Comment (eyalbe4) [inline src/main/utils/mavenUtils.ts:72]: Let's add a description for *nodeCanReachPom*.

File: src/main/utils/mavenUtils.ts [hunk 6]
```diff
@@ -73,6 +69,33 @@ export class MavenUtils {
...
+     * Get <dependency>...</dependency> tag from the pom.xml.
+     * @param pomXmlContent - The pom.xml content
+     * @param groupId - The dependency's group ID
```
Comment (eyalbe4) [inline src/main/utils/mavenUtils.ts:75]: Let's add a description explaining what *ContextKeys.EXCLUDE_DEPENDENCY_ENABLED* does to the node.

File: src/main/utils/mavenUtils.ts (hunk 5 above)
Comment (yahavi) [reply src/main/utils/mavenUtils.ts:72]: Adding JS Doc to `isPomReachable` function

**Pull request merged**